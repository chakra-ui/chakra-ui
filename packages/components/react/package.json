{
  "name": "@chakra-ui/react",
  "version": "2.4.9",
  "description": "Responsive and accessible React UI components built with React and Emotion",
  "main": "src/index.ts",
  "files": [
    "dist"
  ],
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@chakra-ui/accordion": "workspace:*",
    "@chakra-ui/alert": "workspace:*",
    "@chakra-ui/avatar": "workspace:*",
    "@chakra-ui/breadcrumb": "workspace:*",
    "@chakra-ui/button": "workspace:*",
    "@chakra-ui/card": "workspace:*",
    "@chakra-ui/checkbox": "workspace:*",
    "@chakra-ui/close-button": "workspace:*",
    "@chakra-ui/control-box": "workspace:*",
    "@chakra-ui/counter": "workspace:*",
    "@chakra-ui/css-reset": "workspace:*",
    "@chakra-ui/editable": "workspace:*",
    "@chakra-ui/form-control": "workspace:*",
    "@chakra-ui/hooks": "workspace:*",
    "@chakra-ui/icon": "workspace:*",
    "@chakra-ui/image": "workspace:*",
    "@chakra-ui/input": "workspace:*",
    "@chakra-ui/layout": "workspace:*",
    "@chakra-ui/live-region": "workspace:*",
    "@chakra-ui/media-query": "workspace:*",
    "@chakra-ui/menu": "workspace:*",
    "@chakra-ui/modal": "workspace:*",
    "@chakra-ui/number-input": "workspace:*",
    "@chakra-ui/pin-input": "workspace:*",
    "@chakra-ui/popover": "workspace:*",
    "@chakra-ui/popper": "workspace:*",
    "@chakra-ui/portal": "workspace:*",
    "@chakra-ui/progress": "workspace:*",
    "@chakra-ui/provider": "workspace:*",
    "@chakra-ui/radio": "workspace:*",
    "@chakra-ui/react-env": "workspace:*",
    "@chakra-ui/select": "workspace:*",
    "@chakra-ui/skeleton": "workspace:*",
    "@chakra-ui/slider": "workspace:*",
    "@chakra-ui/spinner": "workspace:*",
    "@chakra-ui/stat": "workspace:*",
    "@chakra-ui/styled-system": "workspace:*",
    "@chakra-ui/switch": "workspace:*",
    "@chakra-ui/system": "workspace:*",
    "@chakra-ui/table": "workspace:*",
    "@chakra-ui/tabs": "workspace:*",
    "@chakra-ui/tag": "workspace:*",
    "@chakra-ui/textarea": "workspace:*",
    "@chakra-ui/theme": "workspace:*",
    "@chakra-ui/theme-utils": "workspace:*",
    "@chakra-ui/toast": "workspace:*",
    "@chakra-ui/tooltip": "workspace:*",
    "@chakra-ui/transition": "workspace:*",
    "@chakra-ui/utils": "workspace:*",
    "@chakra-ui/visually-hidden": "workspace:*"
  },
  "peerDependencies": {
    "@emotion/react": "^11.0.0",
    "@emotion/styled": "^11.0.0",
    "framer-motion": ">=4.0.0",
    "react": ">=18",
    "react-dom": ">=18"
  },
  "author": "Segun Adebayo <sage@adebayosegun.com>",
  "license": "MIT",
  "homepage": "https://chakra-ui.com/",
  "repository": {
    "type": "git",
    "url": "https://github.com/chakra-ui/chakra-ui",
    "directory": "packages/components/react"
  },
  "keywords": [
    "react",
    "ui",
    "design-system",
    "react-components",
    "uikit",
    "accessible",
    "components",
    "emotion",
    "library",
    "design-system"
  ],
  "storybook": {
    "title": "Chakra UI",
    "url": "https://storybook.chakra-ui.com"
  },
  "devDependencies": {
    "@emotion/react": "^11.9.0",
    "@emotion/styled": "^11.8.1",
    "framer-motion": "^6.2.9",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "clean-package": "2.1.1"
  },
  "scripts": {
    "build": "tsup src --dts",
    "dev": "pnpm build:fast -- --watch",
    "clean": "rimraf dist .turbo",
    "typecheck": "tsc --noEmit",
    "build:fast": "tsup src",
    "prepack": "clean-package",
    "postpack": "clean-package restore"
  },
  "clean-package": "../../../clean-package.config.json",
  "tsup": {
    "clean": true,
    "target": "es2019",
    "format": [
      "cjs",
      "esm"
    ]
  }
}
