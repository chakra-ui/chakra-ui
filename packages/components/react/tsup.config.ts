import fs from "node:fs"
import path from "node:path"
import { defineConfig } from "tsup"

function removeDirective(fileName: string) {
  const file = path.resolve(__dirname, "dist", fileName)
  return fs.promises.writeFile(
    file,
    fs.readFileSync(file, "utf-8").replace("'use client'\n", ""),
  )
}

export default defineConfig({
  format: ["cjs", "esm"],
  target: "es2019",
  sourcemap: true,
  banner: {
    js: "'use client'",
  },
  async onSuccess() {
    await Promise.all([
      removeDirective("index.mjs"),
      removeDirective("index.js"),
    ])
  },
})
