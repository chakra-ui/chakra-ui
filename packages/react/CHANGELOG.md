# Change Log

## 1.0.3

### Patch Changes

- [`0d0ed051`](https://github.com/chakra-ui/chakra-ui/commit/0d0ed0513ac1094833f1e0294f655af122682ff4)
  [#2707](https://github.com/chakra-ui/chakra-ui/pull/2707) Thanks
  [@TimKolberger](https://github.com/TimKolberger)! - Fix too narrow TypeScript
  type for theme override

- Updated dependencies
  [[`f3cc344e`](https://github.com/chakra-ui/chakra-ui/commit/f3cc344e1a7dd08dc7a312e6ce039d55a80cc526),
  [`72bbd0db`](https://github.com/chakra-ui/chakra-ui/commit/72bbd0dbb913ba38ee2b9191d12bf73713ae4398),
  [`653f3dd6`](https://github.com/chakra-ui/chakra-ui/commit/653f3dd6f30a17e366c069666acbfd9eddb11936),
  [`c73cce25`](https://github.com/chakra-ui/chakra-ui/commit/c73cce25bf986fd5b1261575ca395f731f827f53)]:
  - @chakra-ui/pin-input@1.1.0
  - @chakra-ui/theme@1.2.1
  - @chakra-ui/close-button@1.0.2
  - @chakra-ui/system@1.0.2
  - @chakra-ui/alert@1.0.2
  - @chakra-ui/toast@1.1.1
  - @chakra-ui/modal@1.2.1
  - @chakra-ui/popover@1.0.3
  - @chakra-ui/skeleton@1.0.2

## 1.0.2

### Patch Changes

- [`cfd3b325`](https://github.com/chakra-ui/chakra-ui/commit/cfd3b32564066076529811c5350aff6be565b7a3)
  [#2632](https://github.com/chakra-ui/chakra-ui/pull/2632) Thanks
  [@TimKolberger](https://github.com/TimKolberger)! - Improve typings for
  `extendTheme`. IntelliSense now can suggest property keys for colors and
  existing components.
- Updated dependencies
  [[`e1b095cd`](https://github.com/chakra-ui/chakra-ui/commit/e1b095cdb799bb3630d3cbb7891d8c7e929db0f4),
  [`6b527414`](https://github.com/chakra-ui/chakra-ui/commit/6b52741456f71ba4ec8cfe3e6307796d195adf5b),
  [`28bd21d7`](https://github.com/chakra-ui/chakra-ui/commit/28bd21d793911ba56bd146dd7aaff1008a70d147),
  [`2192be3d`](https://github.com/chakra-ui/chakra-ui/commit/2192be3db77504addffc89cbfef62fb9e884fa5d),
  [`a36ede75`](https://github.com/chakra-ui/chakra-ui/commit/a36ede7519b7193f90e4985636a06c5d483a8a62),
  [`ff10bcec`](https://github.com/chakra-ui/chakra-ui/commit/ff10bceca5774769766eef3a6812a22f387dd58d),
  [`ac0af38d`](https://github.com/chakra-ui/chakra-ui/commit/ac0af38d12dc5828503e178fb94a21f252e3a62b),
  [`c696345a`](https://github.com/chakra-ui/chakra-ui/commit/c696345a711338a23542a7b1911a33927a9ba5f1),
  [`9fdc61d8`](https://github.com/chakra-ui/chakra-ui/commit/9fdc61d8801f6d76783b5c9f068525d4dfc28b20),
  [`2416cf9a`](https://github.com/chakra-ui/chakra-ui/commit/2416cf9abe183a3a38adbccff794088d86a46341),
  [`58e26653`](https://github.com/chakra-ui/chakra-ui/commit/58e26653134caee1dd2caddb9014f7a90dc4eb7e)]:
  - @chakra-ui/number-input@1.0.2
  - @chakra-ui/layout@1.1.0
  - @chakra-ui/modal@1.2.0
  - @chakra-ui/theme@1.2.0
  - @chakra-ui/form-control@1.0.2
  - @chakra-ui/toast@1.1.0
  - @chakra-ui/transition@1.0.2
  - @chakra-ui/radio@1.0.2
  - @chakra-ui/input@1.0.2
  - @chakra-ui/select@1.0.2
  - @chakra-ui/textarea@1.0.2
  - @chakra-ui/accordion@1.0.2
  - @chakra-ui/menu@1.0.2
  - @chakra-ui/popover@1.0.2

## 1.0.1

### Patch Changes

- Updated dependencies
  [[`b3594802`](https://github.com/chakra-ui/chakra-ui/commit/b3594802714115c762013174badd5b838217be6f),
  [`843854ec`](https://github.com/chakra-ui/chakra-ui/commit/843854ec669367623b50a598402be343866d87a8),
  [`f0d03d9f`](https://github.com/chakra-ui/chakra-ui/commit/f0d03d9f6e474a1f1767643889734665c2910dc1),
  [`892ea2ca`](https://github.com/chakra-ui/chakra-ui/commit/892ea2ca1c02b4127f4f044df33de58cc7641f5c),
  [`5c482483`](https://github.com/chakra-ui/chakra-ui/commit/5c482483ce24fc798540c9792a15e06772eae213),
  [`204ff7e3`](https://github.com/chakra-ui/chakra-ui/commit/204ff7e39dd37003786194704b36069ef94d56a6),
  [`ba262ac7`](https://github.com/chakra-ui/chakra-ui/commit/ba262ac7b2e2d932bb227d4ff9181e83fbaa4149),
  [`a1ff404b`](https://github.com/chakra-ui/chakra-ui/commit/a1ff404b12a898ab97af024391a06c34da5bc69a)]:
  - @chakra-ui/layout@1.0.1
  - @chakra-ui/theme@1.1.0
  - @chakra-ui/progress@1.0.1
  - @chakra-ui/tabs@1.0.1
  - @chakra-ui/utils@1.0.1
  - @chakra-ui/hooks@1.0.1
  - @chakra-ui/modal@1.1.0
  - @chakra-ui/media-query@1.0.1
  - @chakra-ui/system@1.0.1
  - @chakra-ui/toast@1.0.1
  - @chakra-ui/accordion@1.0.1
  - @chakra-ui/alert@1.0.1
  - @chakra-ui/avatar@1.0.1
  - @chakra-ui/breadcrumb@1.0.1
  - @chakra-ui/button@1.0.1
  - @chakra-ui/checkbox@1.0.1
  - @chakra-ui/close-button@1.0.1
  - @chakra-ui/control-box@1.0.1
  - @chakra-ui/counter@1.0.1
  - @chakra-ui/editable@1.0.1
  - @chakra-ui/form-control@1.0.1
  - @chakra-ui/icon@1.0.1
  - @chakra-ui/image@1.0.1
  - @chakra-ui/input@1.0.1
  - @chakra-ui/live-region@1.0.1
  - @chakra-ui/menu@1.0.1
  - @chakra-ui/number-input@1.0.1
  - @chakra-ui/pin-input@1.0.1
  - @chakra-ui/popover@1.0.1
  - @chakra-ui/popper@1.0.1
  - @chakra-ui/portal@1.0.1
  - @chakra-ui/radio@1.0.1
  - @chakra-ui/select@1.0.1
  - @chakra-ui/skeleton@1.0.1
  - @chakra-ui/slider@1.0.1
  - @chakra-ui/spinner@1.0.1
  - @chakra-ui/stat@1.0.1
  - @chakra-ui/switch@1.0.1
  - @chakra-ui/tag@1.0.1
  - @chakra-ui/textarea@1.0.1
  - @chakra-ui/tooltip@1.0.1
  - @chakra-ui/transition@1.0.1
  - @chakra-ui/visually-hidden@1.0.1

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.0.0 (2020-11-13)

**Note:** Version bump only for package @chakra-ui/react

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.0.0-rc.8 (2020-10-29)

### Bug Fixes

- **toast:** allow custom render in update
  ([eb8bff9](https://github.com/chakra-ui/chakra-ui/commit/eb8bff911e6ec9de0165ab1e8f5ca10d5e022459)),
  closes [#2362](https://github.com/chakra-ui/chakra-ui/issues/2362)

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.0.0-rc.7 (2020-10-25)

**Note:** Version bump only for package @chakra-ui/core

# 1.0.0-rc.6 (2020-10-25)

**Note:** Version bump only for package @chakra-ui/core

# 1.0.0-rc.5 (2020-09-27)

**Note:** Version bump only for package @chakra-ui/core

# 1.0.0-rc.4 (2020-09-25)

**Note:** Version bump only for package @chakra-ui/core

# 1.0.0-rc.3 (2020-08-30)

**Note:** Version bump only for package @chakra-ui/core

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.0.0-rc.2 (2020-08-09)

**Note:** Version bump only for package @chakra-ui/core

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-rc.1](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@1.0.0-rc.0...@chakra-ui/core@1.0.0-rc.1) (2020-08-06)

### Bug Fixes

- stack key issue and yarn2 deps issue
  ([d6cb6b8](https://github.com/chakra-ui/chakra-ui/commit/d6cb6b8fd964729efdf41b1e29c888a3c101316c))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-rc.0](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@1.0.0-next.7...@chakra-ui/core@1.0.0-rc.0) (2020-07-26)

**Note:** Version bump only for package @chakra-ui/core

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.7](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@1.0.0-next.6...@chakra-ui/core@1.0.0-next.7) (2020-07-26)

**Note:** Version bump only for package @chakra-ui/core

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.6](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@1.0.0-next.5...@chakra-ui/core@1.0.0-next.6) (2020-07-15)

**Note:** Version bump only for package @chakra-ui/core

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.5](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@1.0.0-next.4...@chakra-ui/core@1.0.0-next.5) (2020-07-15)

### Bug Fixes

- revert icon fallback to correct svg path
  ([afa4dbb](https://github.com/chakra-ui/chakra-ui/commit/afa4dbb49931391778f635784bddd3395d555df0))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.4](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.8.0...@chakra-ui/core@1.0.0-next.4) (2020-07-01)

### Bug Fixes

- [#891](https://github.com/chakra-ui/chakra-ui/issues/891)
  ([e107acc](https://github.com/chakra-ui/chakra-ui/commit/e107acc8487898a965b0d695c1da71f46fc56d5e))
- Adding PIN input to core export
  ([301483e](https://github.com/chakra-ui/chakra-ui/commit/301483e47996d266612f4f282611dfffca333bbc))
- ts issue with sx prop
  ([d3b1340](https://github.com/chakra-ui/chakra-ui/commit/d3b1340cb255937927b4d4c56ce218141570b951))
- **core:** add Skeleton export
  ([9255f24](https://github.com/chakra-ui/chakra-ui/commit/9255f24a3653f6f71773d4fdf7221c40aeddb873))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.3](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.8.0...@chakra-ui/core@1.0.0-next.3) (2020-06-28)

### Bug Fixes

- [#891](https://github.com/chakra-ui/chakra-ui/issues/891)
  ([e107acc](https://github.com/chakra-ui/chakra-ui/commit/e107acc8487898a965b0d695c1da71f46fc56d5e))
- ts issue with sx prop
  ([d3b1340](https://github.com/chakra-ui/chakra-ui/commit/d3b1340cb255937927b4d4c56ce218141570b951))
- **core:** add Skeleton export
  ([9255f24](https://github.com/chakra-ui/chakra-ui/commit/9255f24a3653f6f71773d4fdf7221c40aeddb873))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.2](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.8.0...@chakra-ui/core@1.0.0-next.2) (2020-06-21)

### Bug Fixes

- [#891](https://github.com/chakra-ui/chakra-ui/issues/891)
  ([e107acc](https://github.com/chakra-ui/chakra-ui/commit/e107acc8487898a965b0d695c1da71f46fc56d5e))
- **core:** add Skeleton export
  ([9255f24](https://github.com/chakra-ui/chakra-ui/commit/9255f24a3653f6f71773d4fdf7221c40aeddb873))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.4.0](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.3.4...@chakra-ui/core@0.4.0) (2019-10-14)

### Bug Fixes

- **box:** update type definitions
  ([b78723a](https://github.com/chakra-ui/chakra-ui/commit/b78723a))
- **css:** export css function
  ([c8a17c8](https://github.com/chakra-ui/chakra-ui/commit/c8a17c8))
- **cssreset:** remove comments from CSSReset preflight.js
  ([25edb5b](https://github.com/chakra-ui/chakra-ui/commit/25edb5b)), closes
  [#161](https://github.com/chakra-ui/chakra-ui/issues/161)
- **image:** add prop to opt out of the fallbackSrc logic
  ([cd1acc6](https://github.com/chakra-ui/chakra-ui/commit/cd1acc6))
- **image:** add support for native width and height
  ([e4b6165](https://github.com/chakra-ui/chakra-ui/commit/e4b6165))
- **image:** add support for native width and height
  ([298f067](https://github.com/chakra-ui/chakra-ui/commit/298f067))
- improve NumberInput API
  ([bca497e](https://github.com/chakra-ui/chakra-ui/commit/bca497e))
- **image:** improve image hook
  ([4fb5c04](https://github.com/chakra-ui/chakra-ui/commit/4fb5c04))
- **numberinput:** improve NumberInput API
  ([fe0441d](https://github.com/chakra-ui/chakra-ui/commit/fe0441d))
- **numberinput:** update API and docs for numberinput
  ([dc70274](https://github.com/chakra-ui/chakra-ui/commit/dc70274))
- **numberinput:** update story
  ([bda8efc](https://github.com/chakra-ui/chakra-ui/commit/bda8efc))
- **types:** add types for default theme. closes issue
  [#160](https://github.com/chakra-ui/chakra-ui/issues/160)
  ([dfef504](https://github.com/chakra-ui/chakra-ui/commit/dfef504))
- **types:** add types for icons to improve DX
  ([5629bc7](https://github.com/chakra-ui/chakra-ui/commit/5629bc7))
- **types:** update types to allow for migration to TS
  ([73bbe73](https://github.com/chakra-ui/chakra-ui/commit/73bbe73))

### Features

- add SimpleGrid component
  ([ff5760c](https://github.com/chakra-ui/chakra-ui/commit/ff5760c))
- add SimpleGrid component and improve Stack
  ([16cc9ef](https://github.com/chakra-ui/chakra-ui/commit/16cc9ef))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.4](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.3.3...@chakra-ui/core@0.3.4) (2019-10-09)

### Bug Fixes

- **build:** remove babel-runtime dep to fix modal issue
  ([f4ae47f](https://github.com/chakra-ui/chakra-ui/commit/f4ae47f))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.3](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.3.2...@chakra-ui/core@0.3.3) (2019-10-09)

### Bug Fixes

- **box:** add overflow types from styled-system
  ([805cb61](https://github.com/chakra-ui/chakra-ui/commit/805cb61))
- **box:** update type definition for borderWidth props
  ([f4310ec](https://github.com/chakra-ui/chakra-ui/commit/f4310ec))
- **button:** fix the types for leftIcon and rightIcon
  ([9f3ba3a](https://github.com/chakra-ui/chakra-ui/commit/9f3ba3a))
- **cssreset:** update the type definition
  ([d8f2701](https://github.com/chakra-ui/chakra-ui/commit/d8f2701))
- **iconbutton:** fix the types for the iconbutton
  ([b9462d4](https://github.com/chakra-ui/chakra-ui/commit/b9462d4))
- **modal:** add support for preserving scrollbar gap
  ([bd67ea0](https://github.com/chakra-ui/chakra-ui/commit/bd67ea0))
- **types:** export named types in all components
  ([7378b77](https://github.com/chakra-ui/chakra-ui/commit/7378b77))
- resolve yarn v2 pnp issues
  ([0e0a04f](https://github.com/chakra-ui/chakra-ui/commit/0e0a04f))
- **types:** include named exports from theme in main type definitions
  ([68e06d8](https://github.com/chakra-ui/chakra-ui/commit/68e06d8))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.2](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.3.1...@chakra-ui/core@0.3.2) (2019-09-26)

### Bug Fixes

- **alertdialog:** update type definitions
  ([da3c706](https://github.com/chakra-ui/chakra-ui/commit/da3c706))
- **drawer:** update type definitions
  ([290e82c](https://github.com/chakra-ui/chakra-ui/commit/290e82c))
- **input:** resolve the error and focus border color props
  ([45b0d97](https://github.com/chakra-ui/chakra-ui/commit/45b0d97))
- **popover:** update type definitions
  ([3a221a3](https://github.com/chakra-ui/chakra-ui/commit/3a221a3))
- **tooltip:** fix the shouldwrapchildren issues. fixes issue
  [#121](https://github.com/chakra-ui/chakra-ui/issues/121)
  ([5d90b77](https://github.com/chakra-ui/chakra-ui/commit/5d90b77))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.1](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.3.0...@chakra-ui/core@0.3.1) (2019-09-25)

### Bug Fixes

- aliases and forwarding
  ([cec87c6](https://github.com/chakra-ui/chakra-ui/commit/cec87c6))
- compose extraConfig
  ([07c9c2a](https://github.com/chakra-ui/chakra-ui/commit/07c9c2a))
- resolve npm pack issue
  ([8b0676e](https://github.com/chakra-ui/chakra-ui/commit/8b0676e))
- resolve npm pack issue
  ([44b299f](https://github.com/chakra-ui/chakra-ui/commit/44b299f))

# 0.3.0-beta (2019-09-25)

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.2.8...@chakra-ui/core@0.3.0) (2019-09-24)

### Features

- Add typings for Slider.onChange
  ([2f15c6a](https://github.com/chakra-ui/chakra-ui/commit/2f15c6a)), closes
  [chakra-ui/chakra-ui#88](https://github.com/chakra-ui/chakra-ui/issues/88)
- **modal, drawer, alertdialog:** remove console errors and improve api
  ([c5ce4b4](https://github.com/chakra-ui/chakra-ui/commit/c5ce4b4))
