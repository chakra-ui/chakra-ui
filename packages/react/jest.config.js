const baseConfig = require("../../jest.config")

module.exports = {
  ...baseConfig,
}
