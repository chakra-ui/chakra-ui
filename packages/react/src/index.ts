export * from "./collection"
export * from "./components"
export { createContext } from "./create-context"
export * from "./hooks"
export { mergeProps } from "./merge-props"
export { mergeRefs } from "./merge-refs"
export { defaultConfig, defaultSystem } from "./preset"
export { defaultBaseConfig } from "./preset-base"
export * from "./styled-system"
