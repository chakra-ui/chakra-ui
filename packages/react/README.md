# Welcome to Chakra UI ⚡️

[![All Contributors](https://img.shields.io/badge/all_contributors-2-orange.svg?style=flat-square)](#contributors-)

- Works out of the box. Chakra UI contains a set of polished React components
  that work out of the box.

- Flexible & composable. Chakra UI components are built on top of a React UI
  Primitive for endless composability.

- Accessible. Chakra UI components follows the WAI-ARIA guidelines
  specifications.

- Dark Mode 😍: All components are dark mode compatible.

## Looking for the documentation?

https://chakra-ui.com

## Installing Chakra UI

⚡️Chakra UI is made up of multiple components and tools which you can import
one by one. All you need to do is install the `@chakra-ui/react` package:

```sh
$ yarn add @chakra-ui/react
# or
$ npm install --save @chakra-ui/react
```

# Getting set up

To start using the components, please follow these steps:

1. Wrap your application in a `ThemeProvider` provided by **@chakra-ui/react**

```jsx
import { ThemeProvider, ColorModeProvider } from "@chakra-ui/react"

const App = ({ children }) => (
  <ThemeProvider>
    <ColorModeProvider>{children}</ColorModeProvider>
  </ThemeProvider>
)
```

`ColorModeProvider` is a context that provides light mode and dark mode values
to the components. It also comes with a function to toggle between light/dark
mode.

2. Now you can start using components like so!:

```jsx
import { Button } from "@chakra-ui/react"

const App = () => <Button>I just consumed some ⚡️Chakra!</Button>
```

# Contributing

Feel like contributing? That's awesome! We have a
[contributing guide](../../CONTRIBUTING.md) to help guide you.

The components to be built come from the
[Aria Practices Design Patterns and Widgets](https://www.w3.org/TR/wai-aria-practices-1.1).

## Contributors ✨

Thanks goes to these wonderful people
([emoji key](https://allcontributors.org/docs/en/emoji-key)):

<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
<!-- prettier-ignore-start -->
<!-- markdownlint-disable -->
<table>
  <tr>
    <td align="center"><a href="https://github.com/segunadebayo"><img src="https://avatars2.githubusercontent.com/u/6916170?v=4" width="100px;" alt="Segun Adebayo"/><br /><sub><b>Segun Adebayo</b></sub></a><br /><a href="https://github.com/chakra-ui/chakra-ui/commits?author=segunadebayo" title="Code">💻</a> <a href="#maintenance-segunadebayo" title="Maintenance">🚧</a> <a href="https://github.com/chakra-ui/chakra-ui/commits?author=segunadebayo" title="Documentation">📖</a> <a href="#example-segunadebayo" title="Examples">💡</a> <a href="#design-segunadebayo" title="Design">🎨</a></td>
    <td align="center"><a href="https://github.com/tioluwani94"><img src="https://avatars1.githubusercontent.com/u/11310046?v=4" width="100px;" alt="Tioluwani Kolawole"/><br /><sub><b>Tioluwani Kolawole</b></sub></a><br /><a href="https://github.com/chakra-ui/chakra-ui/commits?author=tioluwani94" title="Documentation">📖</a> <a href="#example-tioluwani94" title="Examples">💡</a> <a href="#maintenance-tioluwani94" title="Maintenance">🚧</a></td>
  </tr>
</table>

<!-- markdownlint-enable -->
<!-- prettier-ignore-end -->

<!-- ALL-CONTRIBUTORS-LIST:END -->

This project follows the
[all-contributors](https://github.com/all-contributors/all-contributors)
specification. Contributions of any kind welcome!
