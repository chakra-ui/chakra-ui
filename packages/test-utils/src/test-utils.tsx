import { ChakraProvider } from "@chakra-ui/react"
import "@testing-library/jest-dom/extend-expect"
import {
  render as rtlRender,
  RenderOptions,
  fireEvent,
  RenderResult,
} from "@testing-library/react"
import * as React from "react"
import { toHaveNoViolations, axe } from "jest-axe"
import { createSerializer } from "@emotion/jest"
import { RunOptions } from "axe-core"

expect.addSnapshotSerializer(createSerializer())
expect.extend(toHaveNoViolations)

type UI = Parameters<typeof rtlRender>[0]

// UI-less passthrough fallback to prevent using conditional logic in render
function ChildrenPassthrough({ children }: { children: React.ReactElement }) {
  return children
}

export interface TestOptions extends Omit<RenderOptions, "wrapper"> {
  /**
   * optional additional wrapper, e.g. Context
   *
   * @example
   * ```ts
   * // single wrapper
   * render(<MyConponent />, {
   *  wrapper: MyContext
   * });
   *
   * // multiple wrapper
   * render(<MyConponent />, {
   *  wrapper: ({ children }) => (
   *    <ContextA>
   *      <ContextB>
   *        {children}
   *      <ContextB />
   *    <ContextA />
   *  )
   * });
   *
   * ```
   */
  wrapper?: typeof ChildrenPassthrough
}

/**
 * Custom render for @testing-library/react
 *
 * @see https://testing-library.com/docs/react-testing-library/setup#custom-render
 * @param component the component under test
 * @param options customized test options
 */
export const render = (
  ui: UI,
  { wrapper: Wrapper = ChildrenPassthrough, ...options }: TestOptions = {},
): RenderResult =>
  rtlRender(
    <ChakraProvider>
      <Wrapper>{ui}</Wrapper>
    </ChakraProvider>,
    options,
  )

export { rtlRender }
export { axe }

export * from "@testing-library/react"

export {
  act as invoke,
  renderHook,
  RenderHookOptions,
  RenderHookResult,
} from "@testing-library/react-hooks"

export { default as userEvent } from "@testing-library/user-event"

export const escape = (ui: HTMLElement) =>
  fireEvent.keyDown(ui, { key: "Escape", keyCode: 27 })

type TestA11YOptions = TestOptions & { axeOptions?: RunOptions }

/**
 * Validates against common a11y mistakes.
 *
 * Wrapper for jest-axe
 *
 * @example
 * ```jsx
 * it('passes a11y test', async () => {
 *  await testA11Y(<MyComponent />, options);
 * });
 *
 * // sometimes we need to perform interactions first to render conditional UI
 * it('passes a11y test when open', async () => {
 *  const { container } = render(<MyComponent />, options);
 *
 *  fireEvent.click(screen.getByRole('button'));
 *
 *  await testA11Y(container, options);
 * });
 * ```
 *
 * @see https://github.com/nickcolley/jest-axe#testing-react-with-react-testing-library
 */
export const testA11y = async (
  ui: UI | Element,
  { axeOptions, ...options }: TestA11YOptions = {},
) => {
  const container = React.isValidElement(ui)
    ? render(ui, options).container
    : ui

  const results = await axe(container, axeOptions)

  expect(results).toHaveNoViolations()
}
