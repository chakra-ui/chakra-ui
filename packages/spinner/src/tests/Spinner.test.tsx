import * as React from "react"
import { render } from "@chakra-ui/test-utils"
import { Spinner } from ".."

test("Spinner renders correctly", () => {
  const { asFragment } = render(<Spinner />)
  expect(asFragment()).toMatchSnapshot()
})
