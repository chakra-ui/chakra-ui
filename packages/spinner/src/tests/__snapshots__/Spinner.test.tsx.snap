// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1nibdme"
  >
    <span
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
    >
      Loading...
    </span>
  </div>
</DocumentFragment>
`;
